CRM/Price/BAO/LineItem.php

```diff
@@ -72,6 +72,15 @@ public static function create(&$params) {
         civicrm_api3('MembershipPayment', 'create', $membershipPaymentParams);
       }
     }
+    if ($lineItemBAO->entity_table === 'civicrm_participant' && $lineItemBAO->contribution_id && $lineItemBAO->entity_id) {
+      $participantPaymentParams = [
+        'participant_id' => $lineItemBAO->entity_id,
+        'contribution_id' => $lineItemBAO->contribution_id,
+      ];
+      if (!civicrm_api3('ParticipantPayment', 'getcount', $participantPaymentParams)) {
+        civicrm_api3('ParticipantPayment', 'create', $participantPaymentParams);
+      }
+    }
```

Comment:
@eileenmcnaughton I was thinking to make these parts a bit generalised as I can few common pattern like: 
```php
...
$paymentEntites = CRM_Core_SelectValues::getPaymentEntites();
foreach ($paymentEntites as $entity => $paymentEntity) {
  if ($lineItemBAO->entity_table == 'civicrm_' . $entity && $lineItemBAO->contribution_id && $lineItemBAO->entity_id) {
     $entityParams = [
        $entity . '_id' => $lineItemBAO->entity_id,
        'contribution_id' => $lineItemBAO->contribution_id,
      ];
      if (!civicrm_api3($paymentEntity, 'getcount', $entityParams)) {
         $entityParams['isSkipLineItem'] = TRUE;
        civicrm_api3($paymentEntity, 'create', $entityParams);
      }
  }
}
```

Reply:
but in a separate PR, not blocker for this patch. Overall agree with the change.

Reply:
Yeah - if we get a use case (& tests) we can look at that. I don't want to be generic if there really are just 2 entities